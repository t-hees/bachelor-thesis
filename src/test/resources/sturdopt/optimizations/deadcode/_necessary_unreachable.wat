(module
  (func $main (result i32)
    (block (result i32)
      i32.const 1
      i32.const 1
      i32.eq
      (if 
        (then
          i32.const 87
          br 1
          i32.const 4
          drop
        )
        (else
        )
      )
      i32.const 2
      i32.const 5
      i32.add
    )
  )
  (export "main" (func $main))
)
