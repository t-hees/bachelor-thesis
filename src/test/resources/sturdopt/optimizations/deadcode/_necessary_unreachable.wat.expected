(module
  (func $main (result i32)
    (block (result i32)
      i32.const 1
      i32.const 1
      i32.eq
      drop
      (block
        i32.const 87
        br 1
        unreachable
      )
      unreachable ;; This unreachable is necessary to preserve the correct block signature!
    )
  )
  (export "main" (func $main))
)
